require.js

```diff
@@ -49,10 +49,29 @@
     }
 
     var globalEval = eval; // reassigning causes eval to not use lexical scope.
+    var ArrayPush = Array.prototype.push;
 
     // Non-CommonJS speced extensions should be marked with an "// EXTENSION"
     // comment.
 
+
+	var _Module = function _Module() {};
+	_Module.prototype.id = null;
+	_Module.prototype.display = null;
+	_Module.prototype.require = null;
+	_Module.prototype.factory = void 0;
+	_Module.prototype.exports = void 0;
+	_Module.prototype.redirect = void 0;
+	_Module.prototype.location = null;
+	_Module.prototype.directory = null;
+	_Module.prototype.injected = false;
+	_Module.prototype.mappingRedirect = void 0;
+	_Module.prototype.type = null;
+	_Module.prototype.text = void 0;
+	_Module.prototype.dependees = null;
+	_Module.prototype.extraDependencies = void 0;
+	_Module.prototype.uuid = null;
```

Comment:
It does a lookup on prototype rater than owned properties on each object, so should be more memory efficient, right? Especially if some never get used for the life cycle of an object

> On Oct 6, 2015, at 19:25, Kris Kowal <notifications@github.com> wrote:
> 
> In require.js:
> 
> > +	var _Module = function _Module() {};
> > +	_Module.prototype.id = null;
> > +	_Module.prototype.display = null;
> > +	_Module.prototype.require = null;
> > +	_Module.prototype.factory = void 0;
> > +	_Module.prototype.exports = void 0;
> > +	_Module.prototype.redirect = void 0;
> > +	_Module.prototype.location = null;
> > +	_Module.prototype.directory = null;
> > +	_Module.prototype.injected = false;
> > +	_Module.prototype.mappingRedirect = void 0;
> > +	_Module.prototype.type = null;
> > +	_Module.prototype.text = void 0;
> > +	_Module.prototype.dependees = null;
> > +	_Module.prototype.extraDependencies = void 0;
> > +	_Module.prototype.uuid = null;
> The prototype approach doesn’t help any engine. The constructor approach only helps V8.
> 
> —
> Reply to this email directly or view it on GitHub.
>